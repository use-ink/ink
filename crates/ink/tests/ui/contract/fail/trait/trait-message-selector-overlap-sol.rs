#![allow(unexpected_cfgs)]

#[ink::trait_definition]
pub trait TraitDefinition {
    #[ink(message)]
    fn message(&self);
}

#[ink::contract(abi = "sol")]
pub mod contract {
    use super::TraitDefinition;

    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }

    impl TraitDefinition for Contract {
        #[ink(message)]
        fn message(&self) {}
    }
}

fn main() {}
