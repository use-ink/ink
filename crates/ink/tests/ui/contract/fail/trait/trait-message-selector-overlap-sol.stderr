error[E0119]: conflicting implementations of trait `DispatchableMessageInfo<3793696718>` for type `Contract`
  --> tests/ui/contract/fail/trait/trait-message-selector-overlap-sol.rs:28:9
   |
23 |         pub fn message(&self) {}
   |         ------------------------ first implementation here
...
28 |         fn message(&self) {}
   |         ^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `Contract`
